astropy/coordinates/angles/formats.py

```diff
@@ -541,29 +535,18 @@ def sexagesimal_to_string(values, precision=None, pad=False, sep=(":",), fields=
     elif fields < 2 and values[1] >= 30.0:
         values[0] += 1.0
 
-    literal = []
-    last_value = ""
-    literal.append("{0:0{pad}.0f}{sep[0]}")
+    literal = [f"{np.copysign(values[0], sign):0{pad}.0f}{sep[0]}"]
```

Comment:
In Python strings are immutable, so `literal += <extra_string>` would cause the contents of `literal` to be rewritten multiple times. Having `literal` as a `list` of `str` and calling `"".join(literal)` in the very end should be more efficient. Although the difference in performance might not be critical here, preferring `"".join()` is a good habit.

Reply:
Given that we add at most 3 short strings, the list approach should be slower (and indeed a short test suggests it is). I also think the code with only strings is clearer, which to me is more important.

But given that this had a list, I'm definitely not going to insist - what you did is a clear improvement already.